packages/grafana-ui/src/components/Table/TableCell.tsx

```diff
@@ -1,38 +1,59 @@
 import React, { FC } from 'react';
 import { Cell } from 'react-table';
 import { Field } from '@grafana/data';
+
 import { getTextAlign } from './utils';
-import { TableFilterActionCallback } from './types';
+import { TableCellClickActionCallback, TableFilterActionCallback } from './types';
 import { TableStyles } from './styles';
+import { ClickableTableCell } from './ClickableTableCell';
+import { FilterableTableCell } from './FilterableTableCell';
 
-interface Props {
+export interface Props {
   cell: Cell;
   field: Field;
   tableStyles: TableStyles;
-  onCellClick?: TableFilterActionCallback;
+  onCellClick?: TableCellClickActionCallback;
+  onFilterAdded?: TableFilterActionCallback;
 }
 
-export const TableCell: FC<Props> = ({ cell, field, tableStyles, onCellClick }) => {
+export const TableCell: FC<Props> = ({ cell, field, tableStyles, onCellClick, onFilterAdded }) => {
   const filterable = field.config.filterable;
   const cellProps = cell.getCellProps();
 
-  let onClick: ((event: React.SyntheticEvent) => void) | undefined = undefined;
+  if (cellProps.style) {
+    cellProps.style.textAlign = getTextAlign(field);
+  }
 
   if (filterable && onCellClick) {
-    if (cellProps.style) {
-      cellProps.style.cursor = 'pointer';
-    }
-
-    onClick = () => onCellClick(cell.column.Header as string, cell.value);
+    return (
+      <ClickableTableCell
+        cell={cell}
+        field={field}
+        tableStyles={tableStyles}
+        onCellClick={onCellClick}
+        cellProps={cellProps}
+      />
+    );
```

Comment:
I do not think we need this clickable cell and callback anymore, this was just done as simple version of ad-hoc filters, Explore table should use the onFilterAdded callback instead

Reply:
Gotcha, but this means that Explore also would need to implement both `filter for` and `filter out` functionality.

Reply:
Made changes to Explore to accommodate this new api. Please have a look at changes @kaydelaney @aocenas 🙏